code-analysis/crates/solidity/outputs/typescript/lib/index.d.ts

```diff
@@ -471,23 +471,20 @@ export interface TokenRange {
   start: number
   end: number
 }
-export type CSTRuleNode = CstRuleNode
-export class CstRuleNode {
-  get type(): CSTNodeType.Rule
+export class RuleNode {
+  get type(): NodeType.Rule
   get kind(): Rule
-  children(): (CSTRuleNode | CSTTokenNode | CSTGroupNode)[]
+  children(): (RuleNode | TokenNode | GroupNode)[]
 }
-export type CSTTokenNode = CstTokenNode
-export class CstTokenNode {
-  get type(): CSTNodeType.Token
+export class TokenNode {
+  get type(): NodeType.Token
   get kind(): Token
   lexNode(): LexCharsNode | LexChoiceNode | LexSequenceNode | LexNamedNode
-  trivia(): (CSTRuleNode | CSTTokenNode | CSTGroupNode)[]
+  trivia(): (RuleNode | TokenNode | GroupNode)[]
 }
-export type CSTGroupNode = CstGroupNode
-export class CstGroupNode {
-  get type(): CSTNodeType.Group
-  children(): (CSTRuleNode | CSTTokenNode | CSTGroupNode)[]
+export class GroupNode {
+  get type(): NodeType.Group
+  children(): (RuleNode | TokenNode | GroupNode)[]
 }
 export class Language {
   constructor(version: string)
```

Comment:
Unrelated to this PR:

Since `semver` is an NPM official package, should we accept a parsed/validated `semver::Version` object in the `Language` constructor, instead of the raw string? this makes sure we don't have to worry about version parsing errors, and let the user handle that.

Reply:
Not a trivial change because of how the FFI works i.e. accepting a JS object as a param

Reply:
Right! I was thinking in terms of our wrapper around the native API 😅 
Let's ignore this for now then.